src/resolvers/badges.spec.js

```diff
@@ -0,0 +1,85 @@
+import Factory from '../seed/factories'
+import { GraphQLClient } from 'graphql-request'
+import { host, login } from '../jest/helpers'
+
+const factory = Factory()
+
+describe('report', () => {
+  beforeEach(async () => {
+    await factory.create('User', {
+      email: 'user@example.org',
+      password: '1234'
+    })
+    await factory.create('User', {
+      id: 'u2',
+      name: 'moderator',
+      role: 'moderator',
+      email: 'moderator@example.org'
+    })
+    await factory.create('User', {
+      id: 'u3',
+      name: 'admin',
+      role: 'moderator',
+      email: 'admin@example.org'
+    })
+  })
+
+  afterEach(async () => {
+    await factory.cleanDatabase()
+  })
+
+  const params = {
+    id: 'b1',
+    key: 'indiegogo_en_racoon',
+    type: 'crowdfunding',
+    status: 'permanent',
+    icon: '/img/badges/indiegogo_en_racoon.svg'
+  }
+
+  describe('unauthenticated', () => {
+    let client
+
+    it('throws authorization error', async () => {
+      client = new GraphQLClient(host)
+      let { id, key, type, status, icon } = params
+      await expect(
+        client.request(`mutation {
+        CreateBadge(
+          id: "${id}",
+          key: "${key}",
+          type: ${type},
+          status: ${status},
+          icon: "${icon}"
+        ) { id }
+      }`)
+      ).rejects.toThrow('Not Authorised')
+    })
+
+    describe('authenticated admin', () => {
+      let headers
+      let response
+      let { id, key, type, status, icon } = params
+      beforeEach(async () => {
+        headers = await login({ email: 'admin@example.org', password: '1234' })
+        client = new GraphQLClient(host, { headers })
+        response = await client.request(`mutation {
```

Comment:
Either you could just test that `expect(client.request(...)).resolves.toEqual(expected)` and assume that a record was created or you could query the database again. If you want to query the database you can do so by calling the graphQL endpoint or you could also run cypher directly.

Reply:
In Rails I would do `expect{action}.to(change{Badge.count}.from(0).to(1))` maybe we get the same behaviour somehow?

Reply:
Seems to be possible: https://jestjs.io/docs/en/expect.html#expectextendmatchers

This is how it could look like:
```js
expect(action).toChange({
  check: () => queryOne("MATCH(b:Badge) RETURN count(b)"),
  from: 0,
  to: 1
})
```
The helper function`queryOne` would execute and return the result of a query in one cypher session.